### PR TITLE
LF-3778 Clean up unused translations using i18n script

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This commits the results of running the i18n webapp script with `keepRemoved: false` in `i18next-parser.config.cjs`

As translation work is beginning we want to make sure that our translators are not doing more work than necessary! We also want to check the script output in the upcoming sanity week.

Jira link: https://lite-farm.atlassian.net/browse/LF-3778

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

None of the above -- this should be an invisible change targeting unused lines of translation code.

**How Has This Been Tested?**

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [x] Other (please explain)

Manually reviewed diff. Further testing should happen in context on beta.

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/public/locales/en/translation.json
Comment (SayakaOno) [inline packages/webapp/public/locales/en/translation.json:558]: Are they used in the shared folder? If so, I might want to take some time on reviewing this PR later...
https://github.com/LiteFarmOrg/LiteFarm/blob/95aa1465a0dd3a2ac75324bc2aac3f9381689d2b/packages/shared/validation/sensorCSV.js#L19-L34

Comment (kathyavini) [reply packages/webapp/public/locales/en/translation.json:558]: UGH OH NO! The shared folder!

**Review by antsgar — APPROVED** [review]

**Review by antsgar — COMMENTED** [review]
I had approved but I just saw Sayaka's comments and the merge conflicts so I'll hold off on this one for now 🙂

**Comment by SayakaOno** [comment]
I didn't want to manually fix the conflict, so I hard-reset and re-ran the script.
I will start verifying the changes now.

**Comment by SayakaOno** [comment]
In addition to issues that I found by going through the code, many things are mysteriously broken.

Screenshot 2023-11-08 at 7 56 58 PM
Screenshot 2023-11-08 at 7 55 47 PM
Screenshot 2023-11-08 at 7 58 19 PM
Screenshot 2023-11-08 at 7 49 25 PM
Screenshot 2023-11-08 at 7 37 47 PM
Screenshot 2023-11-08 at 7 39 22 PM
Screenshot 2023-11-08 at 7 49 49 PM
Screenshot 2023-11-08 at 7 56 10 PM
Screenshot 2023-11-08 at 8 05 49 PM
Screenshot 2023-11-08 at 7 41 24 PM
Screenshot 2023-11-08 at 7 49 09 PM
Screenshot 2023-11-08 at 7 56 42 PM
Screenshot 2023-11-08 at 8 08 37 PM
Screenshot 2023-11-08 at 7 51 27 PM

File: packages/webapp/public/locales/en/translation.json
Comment (JimmyRowland) [reply packages/webapp/public/locales/en/translation.json:554]: Probably should fix this in the future. i18n script should be able to recognize them.
I am not sure why translations keys can't be used directly.
If translation keys are calculated t(map[propFromDb]), those translation keys usually goes to a separate translation file

File: packages/webapp/src/locales/i18n.js
Comment (SayakaOno) [inline packages/webapp/src/locales/i18n.js:28]: Some i18n.t() in util files stopped working for some reason, and I had to add namespaces.
https://www.i18next.com/how-to/add-or-load-translations#combined-with-a-backend-plugin

**Comment by SayakaOno** [comment]
I think we are still removing translations in use...

File: packages/webapp/public/locales/en/common.json
Comment (SayakaOno) [inline packages/webapp/public/locales/en/common.json]: `packages/webapp/src/components/Modals/BulkSensorUploadModal/index.jsx`

File: packages/webapp/public/locales/en/translation.json
Comment (SayakaOno) [inline packages/webapp/public/locales/en/translation.json:1487]: `/packages/webapp/src/containers/Finances/CustomRevenueType/CustomRevenueRadios/index.jsx`

File: packages/webapp/src/containers/Finances/index.jsx
Comment (SayakaOno) [inline packages/webapp/src/containers/Finances/index.jsx:101]: Translations were missing. I changed the key to use the existing ones.

Comment (kathyavini) [reply packages/webapp/src/containers/Finances/index.jsx:101]: Ah yes sorry I got the keys confused! I fixed it in this commit two days ago but that PR is not merged. And it actually needs the other key as well so depending on what gets merged first it will conflict again sorry 🙏 

https://github.com/LiteFarmOrg/LiteFarm/pull/2958/commits/3ccb5c64c5e57ac2e9a7577a0b82d0f8583c5e70

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:506]: This one we need, it's part of a dynamically defined string  (I guess the script will always fail with those -- 😔 sad because I just wrote one)

```
t(`ENTITY_TYPES.${entityType.toUpperCase()}`)
```

Screenshot 2023-11-10 at 6 01 19 PM

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:550]: It's used, although I've worked on this flow and I _still_ don't how exactly.

Screenshot 2023-11-10 at 6 10 43 PM

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:634]: Ternary in the call to `t()` breaks script too apparently:

```
{t(
  showZeroAreaWarning
    ? 'FARM_MAP.DRAWING_MANAGER.ZERO_AREA_DETECTED'
    : 'FARM_MAP.DRAWING_MANAGER.ZERO_LENGTH_DETECTED',
)}
```

Screenshot 2023-11-10 at 6 17 33 PM

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [inline packages/webapp/public/locales/en/translation.json:863]: This confused the heck out of me but that is a cool feature of i18n I didn't know about!

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [reply packages/webapp/public/locales/en/translation.json:1228]: That means they are needed, right? I don't remember how to trigger these timed notifications locally to check!

**Review by kathyavini — COMMENTED** [review]
Thank you soooooo much @SayakaOno for going through all of this!! I don't think I would have ever brought up trimming before sanity if I knew that it was going to take this much manual work on top of running the script! We should definitely stick to deleting the translations as soon as we remove from the UI because the script is junk 😂 

I found a few more to preserve on top of the ones you found. I left them as comments but then I realized as the PR opening I can't approve anyway, so maybe I'll just push the changes later (after a break!)

I just wanted to double-check that I understood your notification note though -- they should be preserved, right?

File: packages/webapp/public/locales/en/translation.json
Comment (kathyavini) [reply packages/webapp/public/locales/en/translation.json]: It got added back in just in English? I'll re-delete.

**Comment by kathyavini** [comment]
@antsgar I believe this is ready to review and merge 🙏 The script couldn't detect anything other than explicitly defined translation key strings, so @SayakaOno went through and restored huge chunks of translations that should not have been removed.

I think it's at a good state now where sanity testers can pick up anything else that remains. I can't review but it has my approval! ✅

File: packages/webapp/public/locales/en/translation.json
Comment (SayakaOno) [reply packages/webapp/public/locales/en/translation.json]: I was going to restore it! Thank you!

**Review by SayakaOno — APPROVED** [review]
@kathyavini thank you for double-checking my changes and comments!
I will merge before sanity tests begin!

**Pull request merged**
